lib/capistrano/tasks/assets.rake

```diff
@@ -62,15 +62,28 @@ namespace :deploy do
       end
     end
 
+    def manifest_path(filename)
+      release_path.join('public', fetch(:assets_prefix), filename)
+    end
+
+    def manifest_and_backup_paths
```

Comment:
The problem here is that `manifest_and_backup_paths` method has logic to determine manifest_path and that logic is totally unrelated to backup path. Do not put two different thing in one method.

Also extract a method to just get path for assets_manifest_backup.